server/host_controls.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/pkg/errors"
+)
+
+func (p *Plugin) changeHost(requesterID, channelID, newHostID string) error {
+	if isAdmin := p.API.HasPermissionTo(requesterID, model.PermissionManageSystem); !isAdmin {
+		return errors.New("no permissions to change host")
+	}
+
+	state, err := p.lockCall(channelID)
+	if err != nil {
+		return fmt.Errorf("failed to lock call: %w", err)
+	}
+	defer p.unlockCall(channelID)
+
+	if state == nil || state.Call == nil {
+		return errors.New("no call ongoing")
+	}
+
+	if state.Call.HostID == newHostID {
+		return errors.New("user is already the host")
+	}
```

Comment:
Wondering if this should be a no-op vs error to make it idempotent. Not a big deal.

Reply:
Yep, makes sense, thanks.